microsoft/testsuites/kata_conformance/aks_infra.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import datetime
+import os
+from threading import Lock
+
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.containerservice import ContainerServiceClient
+from azure.mgmt.resource import ResourceManagementClient, SubscriptionClient
+
+from lisa.util import LisaException
+from lisa.util.logger import Logger
+
+global_credential_access_lock = Lock()
+
+
+class AKSInfra:
+    def __init__(
+        self,
+        log: Logger,
+        subscription_id: str,
+        tenant_id: str,
+        client_id: str,
+        client_secret: str,
+    ):
+        self.log = log
+        self._credential_setup(subscription_id, tenant_id, client_id, client_secret)
+        self.kube_path = ""
+
+    def _credential_setup(self, subscription_id, tenant_id, client_id, client_secret):
+        if tenant_id:
```

Comment:
The Azure tools should be in Azure folder, and get credentials from Azure platform. Avoid to put it in test code to reduce dup logics.

Reply:
Taking the credentials separately as case variables allows the testcase to decouple from Azure platform and can be run on bare metal (ready platform) in future.

Reply:
Also, conformance tests can be launched from anywhere but it will always be running on kubernetes cluster (AKS). As Pavan said it can be launched with ready platform as well.

I closed the PR mistakenly while syncing the fork. I am also pushing some new changes with new PR.

Thnx @squirrelsc for the review